### PR TITLE
Misc. fixes for Medtronic Direct

### PR DESCRIPTION
**Pull request by gniezen**
- Sometimes pump returns an empty page with zero header - we now catch this.
- Sometimes there is a CGM reading without SENSOR_TIMESTAMP packet
- Sometimes there is an invalid (stale?) entry in the carb ratios for pump setting changes
- Handles case where temp basal is cancelled at the exact same time a new basal schedule starts

### CODE REVIEW

**Comment by gniezen** [comment]
[[uploader] Medtronic remaining work <dev:5, ux:0, tst:1>](https://trello.com/c/qkp0Mfj6/57-uploader-medtronic-remaining-work-dev-5-ux-0-tst-1)

File: lib/medtronic/processData.js [hunk 1]
```diff
@@ -663,6 +663,10 @@ function getCarbRatios(encoded, units) {
       break;
     }
     var startTime = carbRatio.offset * 30 * sundial.MIN_TO_MSEC;
+    if((carbRatios.length > 0) && (startTime < carbRatios[carbRatios.length-1].start)) {
+      debug('Invalid start time for carb ratio:');
```
Comment (pazaan) [inline lib/medtronic/processData.js]: Was there extra info you wanted to put here after the `:`?

File: lib/medtronic/processData.js [hunk 2]
```diff
@@ -472,7 +472,7 @@ function buildBGRecords(records) {
 function buildCGMRecords(events) {
 
   var postrecords = [];
-  var start = null;
+  var start = events[events.length-1].jsDate;
```
Comment (pazaan) [inline lib/medtronic/processData.js]: Are all events guaranteed to have a `jsDate`?

File: lib/medtronic/processData.js (hunk 1 above)
Comment (gniezen) [inline lib/medtronic/processData.js]: Ah, good catch!

File: lib/medtronic/processData.js (hunk 2 above)
Comment (gniezen) [inline lib/medtronic/processData.js]: For CGM events, yes. And in this specific instance, putting in a check wouldn't really help much, as `start` will then either be `null` or `undefined`.

File: lib/medtronic/medtronicSimulator.js [hunk 3]
```diff
@@ -103,7 +103,7 @@ exports.make = function(config){
               break;
             }
 
-            if (common.computeMillisInCurrentDay(cancelled) < currSchedule[i].start ) {
+            if (common.computeMillisInCurrentDay(cancelled) <= currSchedule[i].start ) {
```
Comment (gniezen) [inline lib/medtronic/medtronicSimulator.js]: @pazaan I'm adding one more change to this PR. What happened here is that if a temp basal is cancelled at the exact same time that a new basal schedule is starting, this if statement would not be triggered,  hence the adding of `=`. I'm merging already, but ping me if there any concerns.

**Pull request merged**